tools/daily-builds/lp-recipe-build.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+# This file is part of Checkbox.
+#
+# Copyright 2022-2023 Canonical Ltd.
+# Written by:
+#   Sylvain Pineau <sylvain.pineau@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Kicks off a deb recipe build in Launchpad.
+"""
+
+import os
+import sys
+
+from launchpadlib.credentials import Credentials
+from launchpadlib.launchpad import Launchpad
+from lazr.restfulclient.errors import BadRequest
+from argparse import ArgumentParser
+
+
+def main():
+    parser = ArgumentParser('Invoke a recipe build on specified branch')
+    parser.add_argument('project',
+                        help="Unique name of the project")
+    parser.add_argument('--recipe', '-r',
+                        help="Recipe name to build with. If there is only one "
+                             "then that will be used by default, if not then "
+                             "this must be specified.")
+    args = parser.parse_args()
+
+    credentials = Credentials.from_string(os.getenv("LP_CREDENTIALS"))
+    lp = Launchpad(
+        credentials, None, None, service_root='production', version='devel')
+    try:
+        project = lp.projects[args.project]
+    except KeyError:
+        parser.error("{} was not found in Launchpad.".format(args.project))
```

Comment:
I don't think [`ArgumentParser` object](https://docs.python.org/3/library/argparse.html?highlight=argumentparser#argparse.ArgumentParser) has an `error()` method.

Do you want to log this error instead? (or just use a `print()` statement)

Reply:
Good question, it's part of the launchapdlib code haven't touched.
But it does actually https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error